docs/DevelopmentGuide.rst

```diff
@@ -0,0 +1,105 @@
+Development Guide
+=================
+
+Ready to contribute to the project? Here is how to set up a local development environment.
+
+Initial Setup
+#############
+
+1. Fork the `cda-tum/mqt-bench <https://github.com/cda-tum/mqt-bench>`_ repository on GitHub (see https://docs.github.com/en/get-started/quickstart/fork-a-repo).
+
+2. Clone your fork locally
+
+    .. code-block:: console
+
+        $ git clone git@github.com:your_name_here/mqt-bench
+
+
+3. Change into the project directory
+
+    .. code-block:: console
+
+        $ cd mqt-bench
+
+4. Create a branch for local development
+
+    .. code-block:: console
+
+        $ git checkout -b name-of-your-bugfix-or-feature
+
+    Now you can make your changes locally.
+
+5. (Optional, **highly recommended**) Set up a virtual environment
+
+    .. code-block:: console
+
+        $ python3 -m venv venv
+        $ source venv/bin/activate
+
+    .. note::
+
+        If you are using Windows, you can use the following command instead:
+
+        .. code-block:: console
+
+            $ python3 -m venv venv
+            $ venv\Scripts\activate.bat
+
+    Ensure that pip, setuptools, and wheel are up to date:
+
+    .. code-block:: console
+
+        (venv) $ pip install --upgrade pip, setuptools, wheel
+
+
+6. (Optional) Install `pre-commit <https://pre-commit.com/>`_ to automatically run a set of checks before each commit.
+
+    .. code-block:: console
+
+        (venv) $ pipx install pre-commit
+        (venv) $ pre-commit install
+
+    If you use macOS, then pre-commit is in brew, use :code:`brew install pre-commit`.
+
+Building the Python module
+##########################
+
+The recommended way of building the Python module is to perform an editable install using `pip <https://pip.pypa.io/en/stable/>`_.
+
+    .. code-block:: console
+
+        (venv) $ pip install --editable .
+
+The :code:`--editable` flag ensures that changes in the Python code are instantly available without re-running the command.
+
+Running Python Tests
+--------------------
+
+The Python part of the code base is tested by unit tests using the `pytest <https://docs.pytest.org/en/latest/>`_ framework.
+The corresponding test files can be found in the :code:`tests/` directory.
+
+    .. code-block:: console
+
+        (venv) $ pytest
+
+This installs all dependencies for running the tests in an isolated environment, builds the Python package, and then runs the tests.
+The :code:`-r` flag ensures that the environment is reused for subsequent runs.
```

Comment:
This statement seems out of place given that you do not provide the `nox` sessions that these statements describe.
You'd have to be a little more elaborate in the description, e.g., people wanting to test their code are expected to install `mqt-bench` with the `[test]` extra, i.e., 
```bash
pip install -e ".[test]"
```

Reply:
Good catch, I adjusted it accordingly.